src/Fantomas.Tests/NumberOfItemsTests.fs

```diff
@@ -266,3 +260,18 @@ h [
     longValueThatIsALotOfCharactersSoooooLong
 ]
 """
+
+[<Test>]
+let ``long expressions with number of items set to 3 will get split on max line length`` () =
+    formatSourceString false """
+[ longValueThatIsALotOfCharactersSoooooLongAndlongValueThatIsALotOfCharactersSoooooLongAndlongValueThatIsALotOfCharactersSoooooLong; longValueThatIsALotOfCharactersSoooooLong ]
+    """
+        { config with
+              ArrayOrListMultilineFormatter = NumberOfItems
+              MaxArrayOrListNumberOfItems = 3 }
+    |> prepend newline
+    |> should equal """
+[ longValueThatIsALotOfCharactersSoooooLongAndlongValueThatIsALotOfCharactersSoooooLongAndlongValueThatIsALotOfCharactersSoooooLong
```

Comment:
I don't think this is valid behaviour to be honest if the user explicitly says to only going multiline when three items are present in the list, the length of each individual item should not matter.

This would be reported as an inconsistency when put together with
`let x = [ a; b ]`

Reply:
In the original Issue (https://github.com/fsprojects/fantomas/issues/1143), the rule given was

> If the number of elements in the expression is > 1 then force a multiline version. If the number of elements in the expression = 1 then force a single line version, unless exceeding the maximum line length.

If we replace `1` with `n` to account for the configurable cutoff, then the expression should be made multiline if it exceeds the maximum line length.  This is what this test was meant to catch, and why one of the expressions is so excessively long: I wanted to exceed the maximum line length without exceeding the cutoff for number of items (3).  This test fails if we replace the `isShortExpression` call in your above review with `if size > maxSize then fallbackExpression ctx else smallExpression ctx`, which is why I added it.

Reply:
Please update the documentation then because I already forgot that detail from the original issue.
Now it makes more sense why you still wanted the check.
Perhaps `expressionFitsOnRestOfLine` can be used instead.

Reply:
Updated.